tests/Entity/ProcurationRequestTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Tests\AppBundle\Entity;
+
+use AppBundle\Entity\ProcurationProxy;
+use AppBundle\Entity\ProcurationRequest;
+use PHPUnit\Framework\TestCase;
+
+class ProcurationRequestTest extends TestCase
+{
+    public function testFrenchRequestWithFrenchProxy()
+    {
+        $proxy = new ProcurationProxy(null);
+        $proxy->setFrenchRequestAvailable(true);
+        $proxy->setProxiesCount(2);
+
+        $request = new ProcurationRequest();
+        $request->setRequestFromFrance(1);
+
+        $request->process($proxy);
+
+        $this->assertEquals(1, $proxy->getFoundRequests()->count());
+        $this->assertEquals(false, $proxy->isFrenchRequestAvailable());
+        $this->assertEquals(true, $proxy->isForeignRequestAvailable());
+    }
+
+    public function testFrenchRequestWithForeignProxy()
+    {
+        $proxy = new ProcurationProxy(null);
+        $proxy->setForeignRequestAvailable(true);
+        $proxy->setProxiesCount(2);
+        $proxy->setVoteCountry('GB');
+
+        $request = new ProcurationRequest();
+        $request->setRequestFromFrance(1);
+        $request->setVoteCountry('GB');
+
+        $request->process($proxy);
+
+        $this->assertEquals(1, $proxy->getFoundRequests()->count());
+        $this->assertEquals(true, $proxy->isFrenchRequestAvailable());
+        $this->assertEquals(true, $proxy->isForeignRequestAvailable());
+    }
+
+    public function testForeignRequestWithFrenchProxy()
+    {
+        $proxy = new ProcurationProxy(null);
+        $proxy->setProxiesCount(2);
+        $proxy->setFrenchRequestAvailable(true);
+        $proxy->setVoteCountry('FR');
+
+        $request = new ProcurationRequest();
+        $request->setRequestFromFrance(0);
+        $request->setVoteCountry('FR');
+
+        $request->process($proxy);
+
+        $this->assertEquals(1, $proxy->getFoundRequests()->count());
+        $this->assertEquals(false, $proxy->isFrenchRequestAvailable());
+        $this->assertEquals(true, $proxy->isForeignRequestAvailable());
+    }
+
+    public function testForeignRequestWithForeignProxy()
+    {
+        $proxy = new ProcurationProxy(null);
+        $proxy->setForeignRequestAvailable(true);
+        $proxy->setVoteCountry('GB');
+        $proxy->setProxiesCount(2);
+
+        $request = new ProcurationRequest();
+        $request->setRequestFromFrance(0);
+        $request->setVoteCountry('GB');
+
+        $request->process($proxy);
+
+        $this->assertEquals(1, $proxy->getFoundRequests()->count());
+        $this->assertEquals(true, $proxy->isFrenchRequestAvailable());
+        $this->assertEquals(true, $proxy->isForeignRequestAvailable());
+    }
+
+    public function testProcessAndUnProcessWithOneProxyCount()
+    {
+        $proxy = new ProcurationProxy(null);
```

Comment:
Just a first pre-review comment, for your all tests, don't hesitate to use a DataProvider to refactor and use easly the expected values